src/components/Modal/src/ModalLayer.vue

```diff
@@ -1,24 +1,37 @@
 <template>
-	<m-transition-spring-up>
-		<div
-			v-if="modalApi.state.vnode"
-			:class="$s.ModalLayer"
-		>
-			<pseudo-window
-				body
-				:class="$s.disableScroll"
+	<div>
+		<m-transition-fade>
+			<div
```

Comment:
I'm thinking we should start to get an overlay component wired up to accept click/tap events to close anything inside of it.

Reply:
This is something else I debated about adding. If we add a listener for overlay / off-modal clicks that behavior would **only** be available on desktops and to desktop users, and since most devs develop on desktops they might add something important to the function of the modal to the overlay click handler without realizing they're leaving out all mobile users. Then again, this may be just another thing we leave as the responsibility of the dev.

Reply:
Is the debate if the overlay component should be internal, external (so dev can choose to use it) or exist at all?  It seems like we should provide something, but worth discussing how.

Reply:
Can we add a click to close behavior?  Some props I saw somewhere else:


closeOnEsc | boolean | If `true`, the modal will close when the `Esc` key is pressed | true
closeOnOverlayClick | boolean | If `true`, the modal will close when the overlay is clicked | true

Reply:
not clear if:
- these would go on ModalLayer or on Modal or both (not sure that last one even makes sense?)
- these should be multiple props or a single props that takes multiple arguments via an array
- should be default enabled and disabled by props or default disabled and enabled by props
- should be cancellable via callback

The good news: we don't have to answer any of this questions in this PR, regardless what the answer is to any of the above questions we can add it later in a follow up PR without introducing any breaking changes.